bindings/java/src/org/sleuthkit/datamodel/SleuthkitCase.java

```diff
@@ -5938,6 +6041,189 @@ private static void closeStatement(Statement statement) {
 	}
 
 	/**
+	 * Sets the end date for the given ingest job
+	 *
+	 * @param ingestJobId The ingest job to set the end date for
+	 * @param endDateTime The end date
+	 *
+	 * @throws TskCoreException If inserting into the database fails
+	 */
+	void setIngestJobEndDateTime(long ingestJobId, long endDateTime) throws TskCoreException {
+		CaseDbConnection connection = connections.getConnection();
+		acquireSharedLock();
+		try {
+			Statement statement = connection.createStatement();
+			statement.executeUpdate("UPDATE ingest_jobs SET end_date_time=" + endDateTime + " WHERE ingest_job_id=" + ingestJobId + ";");
+		} catch (SQLException ex) {
+			throw new TskCoreException("Error updating the end date.", ex);
+		} finally {
+			connection.close();
+			releaseSharedLock();
+		}
+	}
+
+	void setIngestJobStatus(long ingestJobId, IngestJobStatusType status) throws TskCoreException {
+		CaseDbConnection connection = connections.getConnection();
+		acquireSharedLock();
+		try {
+			Statement statement = connection.createStatement();
+			statement.executeUpdate("UPDATE ingest_jobs SET status_id=" + status.ordinal() + " WHERE ingest_job_id=" + ingestJobId + ";");
+		} catch (SQLException ex) {
+			throw new TskCoreException("Error updating the end date.", ex);
+		} finally {
+			connection.close();
+			releaseSharedLock();
+		}
+	}
+
+	/**
+	 * Adds the given ingest job to the database
+	 *
+	 * @param dataSource    The datasource the ingest job is being run on
+	 * @param hostName      The name of the host
+	 * @param ingestModules The ingest modules being run during the ingest job
+	 * @param jobStart      The time the job started
+	 *
+	 * @return An information object representing the ingest job added to the
+	 *         database.
+	 *
+	 * @throws TskCoreException If adding the job to the database fails.
+	 */
+	public final IngestJobInfo addIngestJob(Content dataSource, String hostName, List<IngestModuleInfo> ingestModules, Date jobStart) throws TskCoreException {
```

Comment:
Looking at this with fresh eyes, we should future-proof this API in two ways, so there is little chance of having to deprecate it.
- Add a settingsDir argument now, so we don't have to add it later
- Add jobEnd and status arguments so this could be used in other contexts, not just the current use case of a brand new job where we set the status and date to hard-coded values. I'm willing to make the client work a little harder to avoid future deprecation or overloading.